### ISSUE TITLE
Add utility function for converting geotiffs to area definitions

### ISSUE THREAD

**Issue by davidh-ssec** [pytroll/pyresample#92]
Users some time want to get an area definition for a geotiff file. This isn't a difficult thing to do necessarily, but is easy to get wrong. @loerum has already added this type of functionality to mipp for reading SARS data, but it could likely be rewritten or made more flexible: https://github.com/pytroll/mipp/blob/master/mipp/read_geotiff.py

I think if the function just takes a gdal object `gdal.Open(fn)` and uses the GDAL API for getting the projection information it may work for more than just geotiff.

**Comment by drnextgis** [comment]
Hi @djhoese I'm on it.

**Comment by pnuu** [comment]
We already have it, although it needs also the CRS object from `rasterio`: https://github.com/pytroll/satpy/blob/master/satpy/readers/generic_image.py#L103

**Comment by djhoese** [comment]
@drnextgis Sounds good. I think you could borrow the logic that @pnuu links to, but I'd still like this type of utility in pyresample itself. Some of the logic for this will overlap with what is being done in #138 but that's ok since that PR isn't ready quite yet.

I'm assuming you are trying to get hacktoberfest PRs?

I think the utility function should accept either a filename, a gdal object, or a rasterio object. If a filename is specified then import rasterio and open it. Then it will do whatever calculations are necessary to generate an AreaDefinition.

**Comment by drnextgis** [comment]
Yes I found this issue by `hackctoberfest` tag. But it is a good opportunity to know more about pyresample.

**Comment by drnextgis** [comment]
In example provided by @pnuu the rotation of source raster is not taken into account. But I have questions about this. For example rasterio's `bounds` method takes into account rotation https://github.com/mapbox/rasterio/issues/1420 (fixed). Is it expected by `AreaDefinition` constructor that `area_extent` parameter calculated with taking into account a rotation? I see at least two possible option, but maybe I miss something:

* pass to `area_extent` bbox calculated by rasterio + [affine](https://pypi.org/project/affine/) rotation to `rotation` argument
* pass to `area_extent` and `rotation` values of affine of source image

What is the right approach?

**Comment by drnextgis** [comment]
Example of rotated raster?

**Comment by djhoese** [comment]
Sorry, I misunderstood your question. The `rotation` argument should be the degrees of rotation:

```
    rotation: float
        rotation in degrees (negative is cw)
```

So I guess you could calculate that from the affine rotation. I don't use a lot of rotated area definitions so I can speak on the best way to do this. @mraspaud or @pnuu or @loreclem may have other opinions.

**Comment by drnextgis** [comment]
I've made a PR, please take a look.
